docs/containers/testing-with-docker.md

```diff
@@ -0,0 +1,54 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Testing with docker](#testing-with-docker)
+  - [Using the included docker-compose support](#using-the-included-docker-compose-support)
+    - [Usage](#usage)
+    - [Configuration](#configuration)
+    - [Notes](#notes)
+    - [Example](#example)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Testing with docker
+
+## Using the included docker-compose support
+
+### Usage
+
+```bash
+docker build -t lyft/cartography
+docker-compose run cartography ...
+```
+
+### Configuration
+
+Configuration is possible via the `.compose` directory, which is
+git ignored. Environment for cartography is in
+`.compose/cartography/cartography.env`. neo4j config, logs, etc is
+located at `.compose/neo4j/...`
+
+### Notes
+
+* On initial start of the compose stack, it's necessary to
+change the neo4j user's password through the neo4j UI.
+* Neither the docker image, nor the docker-compose file define an
+entrypoint, so it's necessary to pass in the command being run. This
+also makes it possible to run a custom sync script, rather than only
+cartography.
+
+### Example
+
+`./compose/cartography/cartography.env`:
+```
+# See the cartography github configuration intel module docs
+GITHUB_KEY=BASE64ENCODEDKEY
```

Comment:
I don't know a lot about docker config patterns/practices. Is it possible to avoid asking the developer to save creds in a file?

Reply:
Ah, yeah, I think that can be done. Let me test it and push in a change.